### PR TITLE
Fix magboots not needing a grid to work

### LINKED ISSUE #29029: Mag boots let you walk in space now.
## Description
When you are walking around in space magboots magically walk around like you were on a grid.

**Reproduction**
- mag boots on
- walk around in space away from station.

**Screenshots**

https://github.com/space-wizards/space-station-14/assets/47093363/caff721e-d88d-4590-bccd-44dcb8cfd8fa

**Additional context**
NA

### PR DESCRIPTION
**Pull request by EmoGarbage404**
## About the PR

fixes #29029

keeps the old behavior as a boolean because that's kino.

## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- fix: Magboots no longer work when off-grid.

### CODE REVIEW

**Comment by EmoGarbage404** [comment]
ok this bug is actually really nifty so im proposing a t3 tech

## AntiAnti Gravity boots
magboots that work off-grid. They're battery powered and last for a few minutes.

if you add these i will insta merge.

File: Content.Shared/Clothing/SharedMagbootsSystem.cs [hunk 1]
```diff
@@ -147,6 +147,17 @@ private void OnIsWeightless(Entity<MagbootsComponent> ent, ref InventoryRelayedE
...
+        {
+            var xform = Transform(args.Args.Entity);
+            if (TryComp<GravityComponent>(xform.GridUid, out var gravity) && gravity.Enabled ||
+                TryComp<GravityComponent>(xform.MapUid, out var mapGravity) && mapGravity.Enabled)
+            {
+                return;
+            }
+        }
+
         args.Args.IsWeightless = false;
```
Comment (metalgearsloth) [inline Content.Shared/Clothing/SharedMagbootsSystem.cs:161]: This is called every single tick for mob movement so should really be using entity queries.

Comment (ElectroJr) [reply Content.Shared/Clothing/SharedMagbootsSystem.cs:161]: IsWeightless should probably get cached in the same way as can-move or movement speed modifiers.

**Comment by K-Dynamic** [comment]
> AntiAnti Gravity boots
Just call them Suspense Boots or Grav Boots

**Comment by deltanedas** [comment]
> ok this bug is actually really nifty so im proposing a t3 tech
noob i did first :trollface: https://github.com/space-wizards/space-station-14/issues/29029#issuecomment-2169396362

**Pull request merged**